### PR TITLE
generic IntoAf

### PR DESCRIPTION
**Pull request by burrbull**

### CODE REVIEW

File: src/gpio.rs [hunk 1]
```diff
@@ -691,33 +719,27 @@ macro_rules! af {
...
             ) -> Pin<Gpio, Index, $AFi<PushPull>> {
-                moder.alternate(self.index.index());
-                otyper.push_pull(self.index.index());
-                afr.afx(self.index.index(), $i);
-                self.into_mode()
+                self.into_af_push_pull::<$i>(moder, otyper, afr)
             }
 
             /// Configures the pin to operate as an alternate function open-drain output pin
             pub fn $into_afi_open_drain(
```
Comment (burrbull) [inline src/gpio.rs:728]: Could be deprecated now

**Comment by Sh3Rm4n** [comment]
This looks really cool. Leveraging type inference to save time looking up the correct alternate function number is a great ergonomic boom! 🎉 

I'll have a deeper look later this weekend (hopefully sometime today) but at first glance this looks good

**Review by Sh3Rm4n — APPROVED** [review]
This really looks like straight forward changes and a good improvement overall. Feel free to add the `#[deprecated]`. It might also be ok to remove the old `into_af_...` functions, but this could also wait for another release. 

Thank you very much!

**Comment by burrbull** [comment]
rebased

**Comment by Sh3Rm4n** [comment]
LGTM. Thank you very much!

**Pull request merged**